uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java

```diff
@@ -28,10 +31,191 @@
     @MockBean
     private gameService service;
 
+    @Test
+    public void GET_empty_roomlist() throws Exception {
+        ArrayList<room> roomList = new ArrayList<>();
+        when(service.getRoomList()).thenReturn(roomList);
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"))
+                .andDo(print());
+    }
 
     @Test
-    public void testRoom_POST() throws Exception {
-//        this.mock.perform(post("/Rooms")).andDo(print()).andExpect(status().isOk());
+    public void GET_roomlist() throws Exception {
+        ArrayList<room> roomList = new ArrayList<>();
+        room room = new room("test", false);
+        roomList.add(room);
+        when(service.getRoomList()).thenReturn(roomList);
+
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("test"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room room = new room("test", false);
+        when(service.getRoom("2")).thenReturn(room);
```

Comment:
Can do anyString() instead of a hard coded "2", so it will work in case someone changes the 2 later down for accident.

Reply:
Yessir! I'll change it 👍👍